File: js_port/fs_js/src/funcs/text/IsBlankFunction.js
const { BaseFunction, CallType } = require('../../core/functionBase');
const helpers = require('../helpers');
const { FSDataType } = require('../../core/fstypes');

class IsBlankFunction extends BaseFunction {
  constructor() {
    super();
    this.symbol = 'isblank';
    this.callType = CallType.Prefix;
  }

  get maxParameters() {
    return 1;
  }

  evaluate(provider, parameters) {
    const error = helpers.expectParamCount(this.symbol, parameters, this.maxParameters);
    if (error) {
      return error;
    }

    const value = helpers.ensureTyped(parameters.getParameter(provider, 0));
    if (helpers.typeOf(value) === helpers.FSDataType.Null) {
      return helpers.makeValue(FSDataType.Boolean, true);
    }
    if (helpers.typeOf(value) !== FSDataType.String) {
      return helpers.makeError(helpers.FsError.ERROR_TYPE_MISMATCH, `${this.symbol}: string expected`);
    }
    const text = helpers.valueOf(value);
    return helpers.makeValue(FSDataType.Boolean, text.trim().length === 0);
  }
}

module.exports = {
  IsBlankFunction
};

File: Walya/Functions/Text/IsBlankFunction.cs
using Walya.Core;

namespace Walya.Functions.Text
{
    public class IsBlankFunction : IFsFunction
    {
        public const string SYMBOL = "isBlank";
        public int MaxParsCount => 1;

        public CallType CallType => CallType.Prefix;

        public string Symbol => SYMBOL;

        public int Precedence => 100;

        public object Evaluate(IFsDataProvider parent, IParameterList pars)
        {
            if (pars.Count < 1)
                throw new Error.TypeMismatchError($"{this.Symbol}: argument expected");

            if (pars.GetParameter(parent, 0) == null)
                return true;

            if (pars.GetParameter(parent, 0) is not string str)
                throw new Error.TypeMismatchError($"{this.Symbol}: string expected");

            return string.IsNullOrEmpty(str.Trim());
        }



        public string ParName(int index)
        {
            if (index == 0)
            {
                return "String";
            }
            return "";
        }
    }
}
